### PR TITLE
Issue/3813 Use of ContinuationWrapper in all applicable places. Part #3

### LINKED ISSUE #3813: Use the ContinuationWrapper in all applicable places
We should convert all usage of `Continuation` to [`ContinuationWrapper`](#3773) to prevent the multitude of `IllegalStateException`s.

The following places are affected:

- [x] MediaFileRepository
- [x] LoginNoJetpackRepository
- [x] MagicLinkInterceptorRepository
- [x] OrderDetailRepository
- [x] OrderListRepository
- [x] OrderShipmentProvidersRepository
- [x] GroupedProductListRepository
- [x] ProductDetailRepository
- [x] ProductListRepository
- [x] ProductShippingClassRepository
- [x] ProductCategoriesRepository
- [x] ProductReviewsRepository
- [x] ProductTagRepository
- [x] VariationDetailRepository
- [x] VariationListRepository
- [x] ReviewDetailRepository
- [x] ReviewListRepository

### PR DESCRIPTION
**Pull request by kidinov**
Closes #3813

The PR partially implements changes required for #3813. It contains changes in the following classes:

* ProductTagRepository
* VariationDetailRepository
* VariationListRepository
* ReviewDetailRepository
* ReviewListRepository

# Testing instruction:
### ProductTagRepository
Products -> Product -> Add More details -> Tags

### VariationDetailRepository
Products -> Product with variation -> Variations

### VariationListRepository
Products -> Product with variation -> Variations
 
### ReviewDetailRepository
Reviews -> Open a review

### ReviewListRepository
Reviews

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationDetailRepository.kt [hunk 1]
```diff
@@ -125,16 +109,12 @@ class VariationDetailRepository @Inject constructor(
         if (event.causeOfChange == FETCH_SINGLE_VARIATION &&
             event.remoteProductId == remoteProductId &&
             event.remoteVariationId == remoteVariationId) {
-            if (continuationFetchVariation?.isActive == true) {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationDetailRepository.kt:128]: Similar to [this comment](https://github.com/woocommerce/woocommerce-android/pull/3980#discussion_r628308147) - it seems we are changing behavior of the tracking event. It'd be good to double-check that it's ok to make this change.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt [hunk 2]
```diff
@@ -158,21 +156,23 @@ class ReviewDetailRepository @Inject constructor(
     @Subscribe(threadMode = MAIN)
     fun onProductReviewChanged(event: OnProductReviewChanged) {
         if (event.causeOfChange == FETCH_SINGLE_PRODUCT_REVIEW) {
-            continuationReview?.let {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt:161]: Same as above

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt [hunk 3]
```diff
@@ -181,23 +181,28 @@ class ReviewDetailRepository @Inject constructor(
     @Subscribe(threadMode = MAIN)
     fun onProductChanged(event: OnProductChanged) {
         if (event.causeOfChange == FETCH_SINGLE_PRODUCT && event.remoteProductId == remoteProductId) {
-            continuationProduct?.let {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt:184]: Same as above

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewListRepository.kt [hunk 4]
```diff
@@ -349,38 +354,44 @@ class ReviewListRepository @Inject constructor(
...
                 AnalyticsTracker.track(Stat.NOTIFICATIONS_LOADED)
 
-                continuationNotification?.resume(true)
+                continuationNotification.continueWith(true)
             }
-            continuationNotification = null
         } else if (event.causeOfChange == MARK_NOTIFICATIONS_READ) {
             // Since this can be called from other places, only process this event if we were the
             // one who submitted the request.
-            continuationMarkAllRead?.let {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewListRepository.kt:369]: Same as above

**Review by malinajirka — COMMENTED** [review]
Thanks @kidinov! The changes look good to me but similarly to the previous PR, I'd rather double check that we are okay with the change in behavior of the tracking events.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationDetailRepository.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationDetailRepository.kt:128]: Done

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt:161]: Done

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt (hunk 3 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailRepository.kt:184]: Done

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewListRepository.kt (hunk 4 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewListRepository.kt:369]: Done

**Review by malinajirka — APPROVED** [review]
LGTM! Thanks so much for tackling all the VMs in this PR series 🥇 !

**Pull request merged**